extensions/panache/hibernate-orm-panache-kotlin/deployment/src/main/java/io/quarkus/hibernate/orm/panache/kotlin/deployment/KotlinPanacheCompanionEnhancer.java

```diff
@@ -0,0 +1,51 @@
+package io.quarkus.hibernate.orm.panache.kotlin.deployment;
+
+import static io.quarkus.hibernate.orm.panache.kotlin.deployment.KotlinPanacheResourceProcessor.DOTNAME_PANACHE_COMPANION;
+
+import java.lang.reflect.Modifier;
+
+import javax.persistence.Transient;
+
+import org.jboss.jandex.ClassInfo;
+import org.jboss.jandex.DotName;
+import org.jboss.jandex.FieldInfo;
+import org.jboss.jandex.IndexView;
+import org.objectweb.asm.ClassVisitor;
+
+import io.quarkus.gizmo.DescriptorUtils;
+import io.quarkus.panache.common.deployment.EntityField;
+import io.quarkus.panache.common.deployment.EntityModel;
+import io.quarkus.panache.common.deployment.MetamodelInfo;
+import io.quarkus.panache.common.deployment.PanacheEntityEnhancer;
+
+public class KotlinPanacheCompanionEnhancer extends PanacheEntityEnhancer<MetamodelInfo<EntityModel<EntityField>>> {
+
+    private static final DotName DOTNAME_TRANSIENT = DotName.createSimple(Transient.class.getName());
+    private final ClassInfo panacheCompanionClassInfo;
+
+    public KotlinPanacheCompanionEnhancer(IndexView index) {
+        super(index, KotlinPanacheResourceProcessor.DOTNAME_PANACHE_ENTITY_BASE);
+        modelInfo = new MetamodelInfo<>();
+        panacheCompanionClassInfo = index.getClassByName(DOTNAME_PANACHE_COMPANION);
+    }
+
+    @Override
+    public ClassVisitor apply(String className, ClassVisitor outputClassVisitor) {
+        return new KotlinPanacheCompanionClassVisitor(className, outputClassVisitor, modelInfo, panacheEntityBaseClassInfo,
+                panacheCompanionClassInfo, indexView.getClassByName(DotName.createSimple(className)));
+    }
+
+    @Override
+    public void collectFields(ClassInfo classInfo) {
```

Comment:
I think this is exactly the same as `KotlinPanacheEntityEnhancer#collectFields`

Reply:
It is, actually, i and I was thinking of removing it.  it's vestigial here, tbh, as kotlinc takes care of the field access for us.

Reply:
:+1: